### ISSUE TITLE
LT01 & ST02 interaction adds a space between coalesce keyword and opening parenthesis

### ISSUE THREAD

**Issue by gofford** [sqlfluff/sqlfluff#6359]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I have a classic unnecessary case statement:
```
case when ended_at is null or date(ended_at) > current_date() then true else false end as is_active,
```

Which is correctly being flagged by ST02:
```
=== [dbt templater] Sorting Nodes...
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
== [my_redacted_model.sql] FAIL
L:  31 | P:  23 | ST02 | Unnecessary CASE statement. Use COALESCE function
                       | instead. [structure.simple_case]
```

Applying `sqlfluff fix` "fixes" the offending case statement with malformed SQL:
```
coalesce (ended_at is null or date(ended_at) > current_date(), false) as is_active,
        
        ^- shouldn't be there
```

which fails a subsequent lint due to the space. Running `sqlfluff` fix twice avoides the error because the offending space is picked up by LT01.

### Expected Behaviour

I would expect ST02 to not add a space between the `coalesce` keyword and the opening parenthesis.

### Observed Behaviour

ST02 adds a space between the `coalesce` keyword and the opening parenthesis.

### How to reproduce

This query reproduces the error for me:

```
with
    subscriptions as (
        select *
        from {{ ref('stg__subscriptions') }}
    ),

    club_subscriptions as (
        select
            case
                when ended_at is null or date(ended_at) > current_date()
                    then true
                else false
            end as is_active
        from subscriptions
    )

select *
from club_subscriptions
```

### Dialect

bigquery, with the dbt templater

### Version

Python: Python 3.10.15

- **sqlfluff**
	```
	"sqlfluff==3.2.3",
    "sqlfluff-templater-dbt==3.2.3",
	```

- **dbt**
	```
    "dbt-adapters==1.7.0",
    "dbt-bigquery==1.8.2",
    "dbt-core==1.8.7",
	```

### Configuration

```
[sqlfluff]
templater = dbt
dialect = bigquery
max_line_length = 120
exclude_rules =
    structure.column_order,
    structure.using,
    references.qualification,
    ambiguous.column_count
warnings = RF04

[sqlfluff:templater:dbt]
project_dir = ./dbt
profiles_dir = ./dbt

[sqlfluff:templater:jinja]
load_macros_from_path = ./dbt/macros

[sqlfluff:indentation]
indented_ctes = true
allow_implicit_indents = true

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = lower

[sqlfluff:rules:capitalisation.identifiers]
capitalisation_policy = lower

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = lower

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = lower

[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = lower

```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by alanmcruickshank** [comment]
Awkwardly this isn't an issue with `ST02`. Let's use the following query as an example:

```sql
select
    case
        when ended_at is null or date(ended_at) > current_date()
        then true else false
    end as is_active
from foo
```

If I fix that with _only_ ST02, then I get this (`sqlfluff fix test.sql -r ST02`):

```sql
select
    coalesce(ended_at is null or date(ended_at) > current_date(), false) as is_active
from foo
```

No problem ✅

HOWEVER, if I run it without restricting to just `ST02` (`sqlfluff fix test.sql`), I get the issue you describe:

```sql
select
    coalesce (ended_at is null or date(ended_at) > current_date(),
    false) as is_active
from foo
```

It appears to be an interaction between LT01 and ST02 for reasons I don't quite understand as if I run (`sqlfluff fix test.sql -r LT01,ST02`) I get:

```sql
select
    coalesce (ended_at is null or date(ended_at) > current_date(), false) as is_active
from foo
```

I suspect this is because the typing of the fixes returned by `ST02` is slightly off.
